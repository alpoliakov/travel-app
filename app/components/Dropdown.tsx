import Link from 'next/link';
import React from 'react';
import { useIntl } from 'react-intl';

import { useAuth } from '../lib/useAuth';

export default function Dropdown() {
  const { user, signOut } = useAuth();
  const { formatMessage: f } = useIntl();
  const img =
    'https://images.unsplash.com/photo-1472099645785-5658abf4ff4e?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=facearea&facepad=2&w=256&h=256&q=80';

  const logOut = () => {
    signOut();
  };

  return (
    <div className="dropdown cursor-pointer p-1 mr-2 flex items-center">
      <div className="mr-3">
        <img className="h-10 w-10 rounded-full" src={user.avatar} alt="avatar" />
      </div>
      <div className="">
        <span className="text-white text-lg font-semibold ml-1">{user.name}</span>
      </div>
      <div className="rounded-md shadow-sm flex">
        <button className="flex justify-center items-center block h-3 w-3 overflow-hidden focus:outline-none">
          <div
            id="arrow"
            className="animate-ping inline-flex h-full w-full rounded-full text-gray-700 opacity-75">
            <svg
              className="h-3 w-3 text-white"
              xmlns="http://www.w3.org/2000/svg"
              viewBox="0 0 20 20"
              fill="currentColor">
              <path
                fillRule="evenodd"
                d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
                clipRule="evenodd"
              />
            </svg>
          </div>
        </button>
        <ul className="dropdown-menu absolute w-48 bg-white rounded-lg shadow-xl mt-8 -ml-48 hidden sm:mr-24 md:mr-32 lg:mr-48 xl:mr-64s">
          <li className="">
            <Link href={`/user/${user._id}`}>
              <a className="rounded-t text-gray-800 hover:bg-gray-600 hover:text-white py-2 px-4 block whitespace-no-wrap">
                {f({ id: 'profile' })}
              </a>
            </Link>
          </li>
          <li className="">
            <Link href={`/user/settings/${user._id}`}>
              <a className="text-gray-800 hover:bg-gray-600 hover:text-white py-2 px-4 block whitespace-no-wrap">
                {f({ id: 'settings' })}
              </a>
            </Link>
          </li>
          <li className="">
            <div
              className="rounded-b text-gray-800 hover:bg-gray-600 hover:text-white py-2 px-4 block whitespace-no-wrap"
              onClick={logOut}
              role="presentation">
              {f({ id: 'signOut' })}
            </div>
          </li>
        </ul>
      </div>
    </div>
  );
}
