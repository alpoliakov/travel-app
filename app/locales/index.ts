export const messages = {
  en: {
    btnLogin: 'Login',
    btnSignUp: 'Sign Up',
    search: 'Search',
    phEmail: 'Enter your e-mail',
    phPass: 'Enter your password',
    createAcc: 'Create an account',
    forgetPass: 'Forgot password ?',
    enter: 'Enter',
    load: 'Loading',
    high: 'Max',
    low: 'Min',
    now: 'Currently',
    pass: 'Password (6 characters minimum):',
    phName: 'Enter your name',
    phAvatar: 'Upload your avatar',
    name: 'Name (3 characters minimum):',
    avatar: 'Avatar:',
    register: 'Sign Up',
    signOut: 'Sign Out',
    errorEmail: 'Invalid email field!',
    errorPass: 'Invalid password field!',
    errorName: 'Invalid username field!',
    profile: 'Profile',
    delete: 'Delete',
    or: 'Or',
    selectFile: 'Select a file',
    hello: 'Hello!',
    modalText: 'To view, you need to log in or register!',
    close: 'Close',
  },
  ru: {
    btnLogin: 'Войти',
    btnSignUp: 'Регистрация',
    search: 'Поиск',
    phEmail: 'Введите ваш e-mail',
    phPass: 'Введите ваш пароль',
    createAcc: 'Создать аккаунт',
    forgetPass: 'Забыли пароль ?',
    enter: 'Ввести',
    load: 'Загрузка',
    high: 'Макс',
    low: 'Мин',
    now: 'Сейчас',
    pass: 'Пароль (6 символов минимум):',
    phName: 'Введите ваше имя',
    phAvatar: 'Загрузите ваш аватар',
    name: 'Имя (3 символа минимум):',
    avatar: 'Аватар:',
    register: 'Регистрация',
    signOut: 'Выйти',
    errorEmail: 'Невалидное поле электронной почты!',
    errorPass: 'Невалидное поле пароля!',
    errorName: 'Невалидное поле имени пользователя!',
    profile: 'Профиль',
    delete: 'Удалить',
    or: 'Или',
    selectFile: 'Выберите файл',
    hello: 'Приветствуем!',
    modalText: 'Для просмотра необходимо авторизоваться или зарегистрироваться!',
    close: 'Закрыть',
  },
  uk: {
    btnLogin: 'Увійти',
    btnSignUp: 'Реєстрація',
    search: 'Пошук',
    phEmail: 'Введіть ваш e-mail',
    phPass: 'Введіть ваш пароль',
    createAcc: 'Створити акаунт',
    forgetPass: 'Забули пароль ?',
    enter: 'Ввести',
    load: 'Завантаження',
    high: 'Макс',
    low: 'Miн',
    now: 'Зараз',
    pass: 'Пароль (6 символів мінімум):',
    phName: 'Введіть ваше им`я',
    phAvatar: 'Завантажте ваш аватар',
    name: 'Ім`я (3 символа мінімум):',
    avatar: 'Аватар:',
    register: 'Реєстрація',
    signOut: 'Вийти',
    errorEmail: 'Невалідне поле електронної пошти!',
    errorPass: 'Невалідне поле пароля!',
    errorName: 'Невалідне поле імені користувача',
    profile: 'Профіль',
    delete: 'Видалити',
    or: 'Або',
    selectFile: 'Виберіть файл',
    hello: 'Вітаємо!',
    modalText: 'Для перегляду потрібно увійти або зареєструватися!',
    close: 'Закрити',
  },
};
