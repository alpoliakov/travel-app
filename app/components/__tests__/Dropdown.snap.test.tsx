import React from 'react';
import { IntlProvider } from 'react-intl';
import renderer from 'react-test-renderer';

import { messages } from '../../locales';
import Dropdown from '../Dropdown';

jest.mock('../../lib/useAuth', () => ({
  useAuth: jest.fn(() => ({ signIn: jest.fn(), user: '', error: '' })),
  AuthProvider: ({ children }) => <div>{children}</div>,
}));

jest.mock('next/router', () => ({
  ...jest.requireActual('next/router'),
  useRouter: () => ({
    route: '/',
    pathname: '',
    query: '',
    asPath: '',
    locale: 'en',
    locales: ['en', 'ru', 'uk'],
  }),
}));

it('Dropdown renders correctly', () => {
  const tree = renderer
    .create(
      <IntlProvider locale="ru" messages={messages['ru']}>
        <Dropdown />)
      </IntlProvider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
