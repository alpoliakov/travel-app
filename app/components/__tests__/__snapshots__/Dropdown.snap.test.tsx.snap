// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown renders correctly 1`] = `
Array [
  <div
    className="dropdown cursor-pointer p-1 mr-2 flex items-center"
  >
    <div
      className="mr-3"
    >
      <img
        alt="avatar"
        className="h-10 w-10 rounded-full"
      />
    </div>
    <div
      className=""
    >
      <span
        className="text-white text-lg font-semibold ml-1"
      />
    </div>
    <div
      className="rounded-md shadow-sm flex"
    >
      <button
        className="flex justify-center items-center block h-3 w-3 overflow-hidden focus:outline-none"
      >
        <div
          className="animate-ping inline-flex h-full w-full rounded-full text-gray-700 opacity-75"
          id="arrow"
        >
          <svg
            className="h-3 w-3 text-white"
            fill="currentColor"
            viewBox="0 0 20 20"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              clipRule="evenodd"
              d="M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z"
              fillRule="evenodd"
            />
          </svg>
        </div>
      </button>
      <ul
        className="dropdown-menu absolute w-48 bg-white rounded-lg shadow-xl mt-8 -ml-48 hidden sm:mr-24 md:mr-32 lg:mr-48 xl:mr-64s"
      >
        <li
          className=""
        >
          <a
            className="rounded-t text-gray-800 hover:bg-gray-600 hover:text-white py-2 px-4 block whitespace-no-wrap"
            href="/user/undefined"
            onClick={[Function]}
            onMouseEnter={[Function]}
          >
            Профиль
          </a>
        </li>
        <li
          className=""
        >
          <a
            className="text-gray-800 hover:bg-gray-600 hover:text-white py-2 px-4 block whitespace-no-wrap"
          >
            Settings
          </a>
        </li>
        <li
          className=""
        >
          <div
            className="rounded-b text-gray-800 hover:bg-gray-600 hover:text-white py-2 px-4 block whitespace-no-wrap"
            onClick={[Function]}
            role="presentation"
          >
            Выйти
          </div>
        </li>
      </ul>
    </div>
  </div>,
  ")",
]
`;
